Signal/src/ViewControllers/InboxTableViewCell.m

```diff
@@ -49,41 +50,56 @@ - (nullable NSString *)reuseIdentifier
     return NSStringFromClass(self.class);
 }
 
-- (void)configureWithThread:(TSThread *)thread contactsManager:(OWSContactsManager *)contactsManager
+- (void)configureWithThread:(TSThread *)thread
+            contactsManager:(OWSContactsManager *)contactsManager
+      blockedPhoneNumberSet:(NSSet<NSString *> *)blockedPhoneNumberSet
 {
-    if (!_threadId || ![_threadId isEqualToString:thread.uniqueId]) {
-        dispatch_async(dispatch_get_main_queue(), ^{
-            self.hidden = YES;
-        });
+    OWSAssert([NSThread isMainThread]);
+    OWSAssert(thread);
+    OWSAssert(contactsManager);
+    OWSAssert(blockedPhoneNumberSet);
+
+    BOOL isBlocked = NO;
+    if (!thread.isGroupThread) {
+        NSString *contactIdentifier = thread.contactIdentifier;
+        isBlocked = [blockedPhoneNumberSet containsObject:contactIdentifier];
     }
 
     NSString *name = thread.name;
     if (name.length == 0 && [thread isKindOfClass:[TSGroupThread class]]) {
         name = NSLocalizedString(@"NEW_GROUP_DEFAULT_TITLE", @"");
     }
-    UIImage *avatar = [OWSAvatarBuilder buildImageForThread:thread contactsManager:contactsManager];
     self.threadId = thread.uniqueId;
-    NSString *snippetLabel = [[DisplayableTextFilter new] displayableText:thread.lastMessageLabel];
+    NSString *snippetLabel = (isBlocked ? NSLocalizedString(@"HOME_VIEW_BLOCKED_CONTACT_CONVERSATION",
+                                              @"A label for conversations with blocked users.")
+                                        : [[DisplayableTextFilter new] displayableText:thread.lastMessageLabel]);
 
     NSAttributedString *attributedDate = [self dateAttributedString:thread.lastMessageDate];
-    NSUInteger unreadCount             = [[TSMessagesManager sharedManager] unreadMessagesInThread:thread];
+    NSUInteger unreadCount = [[TSMessagesManager sharedManager] unreadMessagesInThread:thread];
 
-    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
Moving the fast stuff out of async makes the initial repaint a *little* less painful, woo. 👍